docs/addons/metrics-server.md

```diff
@@ -0,0 +1,129 @@
+# Metrics Server AddOn
+
+[Metrics Server](https://github.com/kubernetes-sigs/metrics-server)  is a scalable, efficient source of container resource metrics for Kubernetes built-in autoscaling pipelines. It is not deployed by default in Amazon EKS clusters. The Metrics Server is commonly used by other Kubernetes add ons, such as the [Horizontal Pod Autoscaler](https://docs.aws.amazon.com/eks/latest/userguide/horizontal-pod-autoscaler.html), [Vertical Autoscaling](https://docs.aws.amazon.com/eks/latest/userguide/vertical-pod-autoscaler.html) or the [Kubernetes Dashboard](https://docs.aws.amazon.com/eks/latest/userguide/dashboard-tutorial.html).
+
+> **Important**: Don't use Metrics Server when you need an accurate source of resource usage metrics or as a monitoring solution.
+
+## Usage
+
+#### **`index.ts`**
+```typescript
+import { MetricsServerAddOn, EksBlueprint }  from '@shapirov/cdk-eks-blueprint';
+
+# Deploy Metrics Server v0.5.0
+const metricServerAddOn = new MetricsServerAddOn('v0.5.0');
+const addOns: Array<ClusterAddOn> = [ metricServerAddOn ];
+
+const app = new cdk.App();
+new EksBlueprint(app, 'my-stack-name', addOns, [], {
+  env: {
+      account: <AWS_ACCOUNT_ID>,
+      region: <AWS_REGION>,
+  },
+});
+```
+
+Once deployed, you can see metrics-server pod in the `kube-system` namespace.
+
+```sh
+$ kubectl get deployments -n kube-system
+
+NAME                                                          READY   UP-TO-DATE   AVAILABLE   AGE
+metrics-server                                                1/1     1            1           20m
+```
+
+## Functionality
+
+1. Deploys the metrics-server for the given version using the [components.yaml](https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml) in `kube-system` namespace.
+
+## Testing with Kubernetes Dashboard
+
+For testing, we will use the [Kubernetes Dashboard](https://docs.aws.amazon.com/eks/latest/userguide/dashboard-tutorial.html) to view CPU and memory metrics of our cluster.
```

Comment:
we gotta make this an addon!

Reply:
since it is an add-on should we update the docs?